src/wasm.rs

```diff
@@ -33,17 +34,19 @@ impl HttpClient for WasmClient {
 
     fn send(&self, req: Request) -> BoxFuture<'static, Result<Response, Self::Error>> {
         let fut = Box::pin(async move {
-            let url = format!("{}", req.uri());
-            let req = fetch::new(req.method().as_str(), &url);
+            let req: fetch::Request = fetch::Request::new(req).await?;
             let mut res = req.send().await?;
 
             let body = res.body_bytes();
-            let mut response = Response::new(Body::from(body));
-            *response.status_mut() = http::StatusCode::from_u16(res.status()).unwrap();
-
+            let mut response =
+                Response::new(http_types::StatusCode::try_from(res.status()).unwrap());
+            response.set_body(Body::from(body));
             for (name, value) in res.headers() {
-                let name: http::header::HeaderName = name.parse().unwrap();
-                response.headers_mut().insert(name, value.parse().unwrap());
+                let name: http_types::headers::HeaderName = name.parse().unwrap();
+                response.insert_header(
+                    &name,
+                    value.parse::<http_types::headers::HeaderValue>().unwrap(),
```

Comment:
Casting `value` to a `String` or `&str` is enough to pass it to `insert_header`. `value.parse()` shouldn't be needed here.

Reply:
Good catch, thanks!